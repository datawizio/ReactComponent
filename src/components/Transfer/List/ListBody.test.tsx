import "jsdom-global/register";
import React from "react";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import ListBodyWrapper from "./ListBody";

const listBodyProps = {
  prefixCls: "list-body",
  filteredItems: [
    {
      key: "a",
      title: "a"
    },
    {
      key: "b",
      title: "b"
    },
    {
      key: "c",
      title: "c"
    }
  ],
  selectedKeys: ["a", "b"],
  loading: false,
  loadData: jest.fn(() => {
    const obj = {
      data: [
        {
          key: "a",
          title: "a"
        },
        {
          key: "b",
          title: "b"
        },
        {
          key: "c",
          title: "c"
        }
      ]
    };
    return obj;
  }),
  onItemSelect: jest.fn()
};

const setUp = (props?) => mount(<ListBodyWrapper {...props} />);

describe("ListBody component", () => {
  let component;
  beforeEach(() => {
    component = setUp(listBodyProps);
  });
  it("ListBody render corectly", () => {
    expect(component).toMatchSnapshot();
  });

  it("ListBody skeleton with filteredItems", () => {
    const wrapper = setUp({
      ...listBodyProps,
      loading: true,
      filteredItems: [
        {
          key: "a",
          title: "a"
        }
      ]
    });
    expect(wrapper.find("SkeletonListItem").prop("count")).toBe(2);
  });

  it("ListBody select item", () => {
    component
      .find(".list-body-content-item")
      .not(".list-body-content-item-checked")
      .first()
      .simulate("click");

    expect(component.prop("onItemSelect")).toBeCalledTimes(1);
  });
});
