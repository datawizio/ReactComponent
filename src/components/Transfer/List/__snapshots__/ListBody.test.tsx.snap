// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBody component ListBody render corectly 1`] = `
<ListBodyWrapper
  filteredItems={
    Array [
      Object {
        "key": "a",
        "title": "a",
      },
      Object {
        "key": "b",
        "title": "b",
      },
      Object {
        "key": "c",
        "title": "c",
      },
    ]
  }
  loadData={[MockFunction]}
  loading={false}
  onItemSelect={[MockFunction]}
  prefixCls="list-body"
  selectedKeys={
    Array [
      "a",
      "b",
    ]
  }
>
  <ListBody
    filteredItems={
      Array [
        Object {
          "key": "a",
          "title": "a",
        },
        Object {
          "key": "b",
          "title": "b",
        },
        Object {
          "key": "c",
          "title": "c",
        },
      ]
    }
    loadData={[MockFunction]}
    loading={false}
    onItemSelect={[MockFunction]}
    prefixCls="list-body"
    selectedKeys={
      Array [
        "a",
        "b",
      ]
    }
  >
    <ul
      className="list-body-content"
    >
      <Memo(ListItem)
        checked={true}
        item={
          Object {
            "key": "a",
            "title": "a",
          }
        }
        key="a"
        onClick={[Function]}
        prefixCls="list-body"
      >
        <li
          className="list-body-content-item list-body-content-item-checked"
          onClick={[Function]}
          title="a"
        >
          <Checkbox
            checked={true}
          >
            <Checkbox
              checked={true}
            >
              <label
                className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
              >
                <Checkbox
                  checked={true}
                  className=""
                  defaultChecked={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  prefixCls="ant-checkbox"
                  style={Object {}}
                  type="checkbox"
                >
                  <span
                    className="ant-checkbox ant-checkbox-checked"
                    style={Object {}}
                  >
                    <input
                      checked={true}
                      className="ant-checkbox-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="ant-checkbox-inner"
                    />
                  </span>
                </Checkbox>
              </label>
            </Checkbox>
          </Checkbox>
          <span
            className="list-body-content-item-text"
          >
            a
          </span>
        </li>
      </Memo(ListItem)>
      <Memo(ListItem)
        checked={true}
        item={
          Object {
            "key": "b",
            "title": "b",
          }
        }
        key="b"
        onClick={[Function]}
        prefixCls="list-body"
      >
        <li
          className="list-body-content-item list-body-content-item-checked"
          onClick={[Function]}
          title="b"
        >
          <Checkbox
            checked={true}
          >
            <Checkbox
              checked={true}
            >
              <label
                className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
              >
                <Checkbox
                  checked={true}
                  className=""
                  defaultChecked={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  prefixCls="ant-checkbox"
                  style={Object {}}
                  type="checkbox"
                >
                  <span
                    className="ant-checkbox ant-checkbox-checked"
                    style={Object {}}
                  >
                    <input
                      checked={true}
                      className="ant-checkbox-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="ant-checkbox-inner"
                    />
                  </span>
                </Checkbox>
              </label>
            </Checkbox>
          </Checkbox>
          <span
            className="list-body-content-item-text"
          >
            b
          </span>
        </li>
      </Memo(ListItem)>
      <Memo(ListItem)
        checked={false}
        item={
          Object {
            "key": "c",
            "title": "c",
          }
        }
        key="c"
        onClick={[Function]}
        prefixCls="list-body"
      >
        <li
          className="list-body-content-item"
          onClick={[Function]}
          title="c"
        >
          <Checkbox
            checked={false}
          >
            <Checkbox
              checked={false}
            >
              <label
                className="ant-checkbox-wrapper"
              >
                <Checkbox
                  checked={false}
                  className=""
                  defaultChecked={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  prefixCls="ant-checkbox"
                  style={Object {}}
                  type="checkbox"
                >
                  <span
                    className="ant-checkbox"
                    style={Object {}}
                  >
                    <input
                      checked={false}
                      className="ant-checkbox-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="ant-checkbox-inner"
                    />
                  </span>
                </Checkbox>
              </label>
            </Checkbox>
          </Checkbox>
          <span
            className="list-body-content-item-text"
          >
            c
          </span>
        </li>
      </Memo(ListItem)>
    </ul>
  </ListBody>
</ListBodyWrapper>
`;
